packages/block-editor/src/store/selectors.js

```diff
@@ -1798,3 +1805,42 @@ export function isBlockHighlighted( state, clientId ) {
 export function areInnerBlocksControlled( state, clientId ) {
 	return !! state.blocks.controlledInnerBlocks[ clientId ];
 }
+
+/**
+ * Returns the clientId for the first 'active' block of a given array of block names.
+ * A block is 'active' if it (or a child) is the selected block.
+ * Returns the first match moving up the DOM from the selected block.
+ *
+ * @param {Object} state Global application state.
+ * @param {string[]} validBlocksNames The names of block types to check for.
+ *
+ * @return {string} The matching block's clientId.
+ */
+export const getActiveBlockIdByBlockNames = createSelector(
```

Comment:
I'm not a huge fan of this name, but I can't think of anything better either 🙃 Maybe we should mark it as __experimental so that people don't start using it in the meantime?

Reply:
Agreed, and sure that makes sense to me!  done